lib/glimmer/tk/widget_proxy.rb

```diff
@@ -551,21 +562,91 @@ def handle_listener(listener_name, &listener)
       def on(listener_name, &listener)
         handle_listener(listener_name, &listener)
       end
-      
+
+      def raise_event(event_name, data = nil)
+        data = YAML.dump(data) if data && !data.is_a?(String)
+        tk.event_generate("<#{event_name}>", data: data)
+      end
+
       def unbind_all
         @listeners&.keys&.each do |key|
           if key.to_s.downcase.include?('command')
             @tk.send(key, '')
           else
-            @tk.bind(key, '')
+            @tk.bind_remove(key)
           end
         end
+        @listeners = nil
       end
-      
+
+      def clear
+        unbind_all
+        children.each(&:destroy)
+        @children = []
+      end
+
       def content(&block)
         Glimmer::DSL::Engine.add_content(self, Glimmer::DSL::Tk::WidgetExpression.new, keyword, *args, &block)
       end
 
+      def window?
+        false
+      end
+
+      def closest_window
+        widget = self
+        widget = widget.parent_proxy until widget.window?
+        widget
+      end
```

Comment:
This behavior already exists in `toplevel_parent_proxy` & `ancestor_proxies`

As a result, please delete this method. But, if you like, you can add `closest_window` as an alias to `toplevel_parent_proxy`